Mathlib/GroupTheory/MaximalSubgroups.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+
+import Mathlib.Algebra.Group.Subgroup.Lattice
+import Mathlib.Order.Atoms
+
+/-! # Maximal subgroups
+
+A subgroup `IsMaximal` if it is maximal in the collection of
+proper subgroups.
+
+We prove a few basic results which are essentially copied from
+those about maximal ideals.
+
+Besides them, we have :
+* `isMaximal_iff` proves that a subgroup K of G is maximal
+if and only  if it is not ⊤ and if for all g ∈ G such that g ∉ K,
+every subgroup containing K and g is ⊤.
+
+## TODO
+
+Is it useful to write it in a greater generality?
+Will we need to write this for all `sub`-structures ?
+
+-/
+
+
+variable {G : Type*} [Group G]
+
+namespace Subgroup
+
+/-- A subgroup is maximal if it is maximal in the collection of proper subgroups. -/
+class _root_.AddSubgroup.IsMaximal {G : Type*} [AddGroup G] (K : AddSubgroup G) : Prop where
+/-- An subgroup is maximal if it is maximal in the collection of proper subgroups. -/
+  out : IsCoatom K
```

Comment:
move this out of the namespace?

Also maybe we shouldn't introduce more `out` as field names in the wake of [this Zulip thread](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/too.20many.20.60out.60s/near/492051754).

Reply:
The rewrite makes this obsolate.